src/Microsoft.AspNet.Owin/OwinExtensions.cs

```diff
@@ -67,21 +68,32 @@ public static IApplicationBuilder UseOwin(this IApplicationBuilder builder, Acti
 
         public static IApplicationBuilder UseBuilder(this AddMiddleware app)
         {
+            return app.UseBuilder(configureServices: null);
+        }
+
+        public static IApplicationBuilder UseBuilder(this AddMiddleware app, Action<IServiceCollection> configureServices)
```

Comment:
The current approach forces us to depend on our implementation of DI. If we let the user return a `IServiceProvider`, they can configure and return a container of their choice. It would also remove the need to depend on `Microsoft.Extensions.DependencyInjection` and instead depend on `Microsoft.Extensions.DependencyInjection.Abstractions`.